lib/mods/clean-console.js

```diff
@@ -1,74 +1,45 @@
 'use strict';
 
-/**
- * Gets the Collection of CallExpressions with callee name console.
- *
- * @param {any} j - the jscodemod api
- * @param {Collection} root - the root Collection
- *
- * @return {Collection} a Collection of console statements
- */
-function getConsoleStatements(j, root) {
-  return root.find(j.CallExpression, {
-    callee: {
-      object: {
-        name: 'console',
-      },
-    },
-  });
-};
+/* eslint-disable require-jsdoc */
+const nodeTypes = require('ts-morph').TypeGuards;
 
 /**
  * Replaces console statements without ExpressionStatement parents
  * with `undefined`.
  *
- * @param {any} j - the jscodemod api
- * @param {Collection} root - the root Collection
- *
- * @return {Collection} the modified root Collection
- */
-function cleanConsoleExpressions(j, root) {
-  getConsoleStatements(j, root).filter((p) => {
-    const parentType = p.parentPath.value.type;
-
-    return parentType !== 'ExpressionStatement';
-  })
-      .replaceWith(() => {
-        return `undefined`;
-      });
-
-  return root;
-};
-
-/**
- * Removes console statements with ExpressionStatement parents.
- *
- * @param {any} j - the jscodemod api
- * @param {Collection} root - the root Collection
+ * @param {Node} node - a typed AST node
  *
- * @return {Collection} the modified root Collection
+ * @return {*} whether the node is a console call expression
  */
-function cleanConsoleStatements(j, root) {
-  getConsoleStatements(j, root).forEach((path) => {
-    if (path.parentPath.value.type === 'ExpressionStatement') {
-      j(path.parentPath).replaceWith(() => {
-        return;
-      });
-    }
-  });
-
-  return root;
+function isConsoleCallExpression(node) {
+  if (!nodeTypes.isCallExpression(node)) {
+    return;
+  }
+
+  const propertyAccess = node.getExpression();
+  if (!nodeTypes.isPropertyAccessExpression(propertyAccess)) {
+    return;
+  }
+
+  const base = propertyAccess.getExpression();
+  const typeName = base.getType().getText();
+  return typeName === 'Console';
 };
 
 /**
  * Removes all console statements and expressions
  *
- * @param {any} j - the jscodemod api
- * @param {Collection} root - the root Collection
+ * @param {Node} node - the root typed AST node
```

Comment:
This doesn't seem accurate. It runs on every node, not just the root.

Reply:
Ah good catch, I was still in the jscodeshift mindset